### PR TITLE
Methods predicate

### LINKED ISSUE #1272: Multiple methods in Method predicate
**Is your feature request related to a problem? Please describe.**
I need to use separate filters for separate HTTP Methods but same path. E.G. I need to modify body of the request only for PUT,POST and PATCH. This will cause configuration duplication as only single method is allowed for predicate.

Example:
```
put_operations: Path("...") && Method("PUT")
 -> modPath("/some-path/", "/another-path")
 -> modifyBody()
 -> "https://google.com";

post_operations: Path("...") && Method("POST")
 -> modPath("/some-path/", "/another-path")
 -> modifyBody()
 -> "https://google.com";

operations: Path("...")
 -> modPath("/some-path/", "/another-path")
 -> "https://google.com";
```

Would become:
```
put_operations: Path("...") && Method("PUT", "POST", "PATCH")
 -> modPath("/some-path/", "/another-path")
 -> modifyBody()
 -> "https://google.com";

operations: Path("...")
 -> modPath("/some-path/", "/another-path")
 -> "https://google.com";
```

There also might be other use cases for that.

**Describe the solution you would like**
Allow `Method` predicate accept an array instead of single value.

**Would you like to work on it?**
Yes, but I might be missing some details, on why it's not already done.

### PR DESCRIPTION
**Pull request by buzzlighty**
This PR adds new Methods predicate which allows filtering on multiple methods.

```
Methods("POST")
Methods("GET", "POST")
```

Resolves #1272

### CODE REVIEW

**Comment by szuecs** [comment]
Thanks for your pull request!

Please resolve staticcheck complains.
Run yourself with `make staticcheck`

predicates/methods/methods.go:79:16: should use fmt.Errorf(...) instead of errors.New(fmt.Sprintf(...)) (S1028)
506Makefile:140: recipe for target 'staticcheck' failed
507make[1]: *** [staticcheck] Error 1

**Comment by buzzlighty** [comment]
@szuecs thanks for pointing. I've rebased branch with static check fix

**Comment by buzzlighty** [comment]
CI says:
```
time="2020-02-16T21:31:16Z" level=error msg="Error while calling token introspection: Post http://client-id:***@127.0.0.1:35089/test-auth: dial tcp 127.0.0.1:35089: connect: connection refused."
```

I think it's temporary, just needed to be retriggered.

File: predicates/methods/methods.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
+/*
+
+Package source implements a custom predicate to match routes
```
Comment (szuecs) [inline predicates/methods/methods.go]: To make godoc happy: `Package source ...` -> `Package methods...`

File: docs/reference/predicates.md [hunk 2]
```diff
@@ -144,6 +144,23 @@ Method("GET")
 Method("OPTIONS")
 ```
 
+## Methods
+
+The HTTP method that the request must match. HTTP methods are one of
+GET, HEAD, PATCH, POST, PUT, DELETE, OPTIONS, CONNECT.
```
Comment (szuecs) [inline docs/reference/predicates.md]: TRACE is missing

**Comment by szuecs** [comment]
everything else lgtm

File: predicates/methods/methods.go (hunk 1 above)
Comment (buzzlighty) [reply predicates/methods/methods.go]: some copy/paste thing 🤦‍♂

File: docs/reference/predicates.md (hunk 2 above)
Comment (buzzlighty) [reply docs/reference/predicates.md]: same here...

**Comment by buzzlighty** [comment]
@szuecs thanks for comments. Fixed everything and rebased to keep history clean.

**Comment by szuecs** [comment]
:+1:

**Comment by aryszka** [comment]
:+1:

**Comment by szuecs** [comment]
Thanks @buzzlighty !

**Comment by buzzlighty** [comment]
@szuecs it seems like this won't work unless predicate is added to this collection:

https://github.com/zalando/skipper/blob/33403a5ef75a541d6e7cce6980897680dfc2151a/skipper.go#L1104

I've tried using predicate and got an error: "predicate not found: Methods".
I thought the predicates somehow resolved automagically  😔

**Comment by szuecs** [comment]
https://github.com/zalando/skipper/releases/tag/v0.11.49 was released automatically

**Pull request merged**